### PR TITLE
apprclient: add context to public API

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4449.

### CODE REVIEW

**Comment by kopiczko** [comment]
Fishing for lgtm.

**Review by rossf7 — APPROVED** [review]
LGTM. Thanks for taking care. <3

**Review by tuommaki — APPROVED** [review]

File: integration/test/basic/basic_test.go [hunk 1]
```diff
@@ -18,7 +19,7 @@ import (
 	"github.com/spf13/afero"
 )
 
-func test_setup(t *testing.T) (*apprclient.Client, *k8sportforward.Tunnel) {
+func testSetup(ctx context.Context, t *testing.T) (*apprclient.Client, *k8sportforward.Tunnel) {
```
Comment (indyfree) [inline integration/test/basic/basic_test.go]: <3

**Review by indyfree — APPROVED** [review]
LGTM!

**Pull request merged**